seonshine_frontend/src/components/organims/restaurantProfileModal/index.tsx

```diff
@@ -0,0 +1,240 @@
+      } else {
+        setUploadError(null);
+      }
+
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setPreviewUrl(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+      try {
+        // will call api upload photo later
+        // const uploadedImageUrl = await uploadToServer(file);
+        // setPreviewUrl(uploadedImageUrl);
+        console.log('upload image api');
+      } catch (error) {
+        console.error('Error uploading image:', error);
+        setUploadError('Error uploading image. Please try again.');
+      }
+    }
+  };
+
+  const fields = [
+    { name: 'user_id', label: 'ID', disabled: true },
+    { name: 'role_id', label: 'Type of User', disabled: true },
+    { name: 'username', label: 'Full name', disabled: false },
+    { name: 'email', label: 'Email', disabled: true },
+    { name: 'weekday', label: 'Assigned date', disabled: true },
+    { name: 'address', label: 'Address', disabled: false },
+    { name: 'phone_number', label: 'Phone Number', disabled: false },
+    { name: 'user_status', label: 'Status', disabled: true },
+  ];
+
+  useEffect(() => {
+    reset({
+      username: restaurant?.username,
+      address: restaurant?.address,
+      phone_number: restaurant?.phone_number,
+    });
+  }, [restaurant]);
+
+  return (
+    <Modal
+      open={isOpen}
+      onClose={handleClose}
+      aria-labelledby="user-profile-modal-title"
+      aria-describedby="user-profile-modal-description"
+    >
+      <Box className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 w-full md:w-2/3 lg:w-2/5 bg-white shadow-xl rounded-lg">
+        <Box className="flex flex-col md:flex-row">
+          <Box className="w-full md:w-1/4 bg-gray-100 flex flex-col items-center rounded-lg p-4 md:p-0">
+            {isLoading ? (
+              <Skeleton
+                height={84}
+                width={84}
+                className="mt-4 md:mt-12"
+              />
+            ) : (
+              <Avatar
+                alt={restaurant?.username}
+                src={previewUrl}
+                className="w-24 h-24 mt-4 md:mt-12"
+              />
+            )}
+            <input
+              accept="image/*"
+              className="hidden"
+              id="upload-photo"
+              type="file"
+              onChange={handleImageChange}
+            />
+            <label htmlFor="upload-photo">
+              <Button
+                component="span"
+                className="hover:bg-green-300 hover:text-white hover:outline-green-300 bg-white text-green-200 font-bold outline outline-2 outline-green-200 mx-4 mt-4 md:mt-8 rounded-xl"
+              >
+                Select Photo
+              </Button>
+            </label>
+            {uploadError && <p className="text-red-500 text-xs m-2">{uploadError}</p>}
+          </Box>
+          <Box className="w-full md:w-3/4 p-4 md:p-16 relative">
+            {!isEditing ? (
```

Comment:
reuse IconButton and just check `{!isEditing ? <EditIcon /> : <EditOutlinedIcon />} `